cdap-cli/src/main/java/co/cask/cdap/cli/command/system/HelpCommand.java

```diff
@@ -207,44 +208,51 @@ private void printPattern(String pattern, PrintStream output, int colWidth, int
   }
 
   /**
-   * Prints the given string with text wrapping at the given column width.
+   * Prints the given string with text wrapping at the given column width, honoring line breaks.
    *
    * @param str the string to print
    * @param output the {@link PrintStream} to write to
    * @param colWidth width of the column
-   * @param prefixSpaces number of spaces as the prefix for each line printed
+   * @param prefixWidth number of spaces as the prefix for each line printed
    */
-  private void wrappedPrint(String str, PrintStream output, int colWidth, int prefixSpaces) {
-    String prefix = Strings.repeat(" ", prefixSpaces);
-    colWidth -= prefixSpaces;
+  private void wrappedPrint(String str, PrintStream output, int colWidth, int prefixWidth) {
+    char lineChar = '\n';
+    String prefix = Strings.repeat(" ", prefixWidth);
 
-    if (str.length() <= colWidth) {
-      output.printf("%s%s", prefix, str);
-      output.println();
-      return;
+    colWidth -= prefixWidth;
+
+    Iterable<String> lines = Splitter.on(lineChar).split(str);
+    for (String line : lines) {
+      format(line, output, colWidth, prefix);
     }
+  }
 
-    int beginIdx = 0;
-    while (beginIdx < str.length()) {
-      int idx;
-      if (beginIdx + colWidth >= str.length()) {
-        idx = str.length();
-      } else {
-        idx = str.lastIndexOf(' ', beginIdx + colWidth);
-      }
+  /**
+   * Prints the given line with text wrapping on spaces at the given column width.
+   * If the line has no spaces within the column width, prints without wrapping.
+   *
+   * @param line the string (without returns) to print
+   * @param output the {@link PrintStream} to write to
+   * @param colWidth width of the column
+   * @param prefixSpaces prefix for each line printed
+   */
+  private void format(String line, PrintStream output, int colWidth, String prefixSpaces) {
+    output.printf(prefixSpaces);
+    int widthLeft = colWidth - prefixSpaces.length();
```

Comment:
I think it is "width remaining" rather than left-right. But I can change the variable name.